From dde534bdf83c224b33a114356b11f2dd729e5f06 Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Sat, 25 May 2024 16:04:46 +0200
Subject: [PATCH] plat: renesas: rz: ddr: fix build error when DDR_PARAM_SETUP
 is not set

Because the new function "ddr_param_setup" is declared inside a header
file, it can't be re-implemented locally as inline.

Change the code to simply gate calling the function by preprocessor
macro, to avoid linker error when it is neither enabled nor implemented.

Fixes: 'plat: renesas: rz: ddr: call new function ddr_param_setup before setup'
Signed-off-by: Josua Mayer <josua@solid-run.com>
---
 plat/renesas/rz/common/drivers/ddr/ddr.c | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/plat/renesas/rz/common/drivers/ddr/ddr.c b/plat/renesas/rz/common/drivers/ddr/ddr.c
index 75c55c299..309a5a273 100755
--- a/plat/renesas/rz/common/drivers/ddr/ddr.c
+++ b/plat/renesas/rz/common/drivers/ddr/ddr.c
@@ -67,12 +67,6 @@ char data_synd[] = {
 #endif
 #endif
 
-#if defined(DDR_PARAM_SETUP) && DDR_PARAM_SETUP
-void __attribute__((weak)) ddr_param_setup(void) {}
-#else
-static inline void ddr_param_setup(void) {}
-#endif
-
 // main
 void ddr_setup(void)
 {
@@ -82,7 +76,9 @@ void ddr_setup(void)
 	uint32_t	tmp;
 	int i;
 
+#if defined(DDR_PARAM_SETUP) && DDR_PARAM_SETUP
 	ddr_param_setup();
+#endif
 
 	INFO("BL2: setup DDR (Rev. %s)\n", ddr_an_version);
 	// Step2 - Step11
-- 
2.35.3

